src/lib.rs

```diff
@@ -987,82 +1037,108 @@ https://bheisler.github.io/criterion.rs/book/faq.html
 
             self.config.measurement_time = dur;
         }
         if matches.is_present("nresamples") {
-            let num_resamples =
-                value_t!(matches.value_of("nresamples"), usize).unwrap_or_else(|e| {
-                    println!("{}", e);
-                    std::process::exit(1)
-                });
+            let num_resamples = matches.value_of_t_or_exit("nresamples");
 
             assert!(num_resamples > 0);
 
             self.config.nresamples = num_resamples;
         }
         if matches.is_present("noise-threshold") {
-            let num_noise_threshold = value_t!(matches.value_of("noise-threshold"), f64)
-                .unwrap_or_else(|e| {
-                    println!("{}", e);
-                    std::process::exit(1)
-                });
+            let num_noise_threshold = matches.value_of_t_or_exit("noise-threshold");
 
             assert!(num_noise_threshold > 0.0);
 
             self.config.noise_threshold = num_noise_threshold;
         }
         if matches.is_present("confidence-level") {
-            let num_confidence_level = value_t!(matches.value_of("confidence-level"), f64)
-                .unwrap_or_else(|e| {
-                    println!("{}", e);
-                    std::process::exit(1)
-                });
+            let num_confidence_level = matches.value_of_t_or_exit("confidence-level");
 
             assert!(num_confidence_level > 0.0 && num_confidence_level < 1.0);
 
             self.config.confidence_level = num_confidence_level;
         }
         if matches.is_present("significance-level") {
-            let num_significance_level = value_t!(matches.value_of("significance-level"), f64)
-                .unwrap_or_else(|e| {
-                    println!("{}", e);
-                    std::process::exit(1)
-                });
+            let num_significance_level = matches.value_of_t_or_exit("significance-level");
 
             assert!(num_significance_level > 0.0 && num_significance_level < 1.0);
 
             self.config.significance_level = num_significance_level;
         }
 
+        // XXX: Comparison functionality should ideally live in 'cargo-criterion'.
+        if matches.is_present("compare") {
+            if self.connection.is_some() {
+                eprintln!(
+                    "Error: tabulating results is not supported when running with cargo-criterion."
+                );
+                std::process::exit(1);
+            }
+            // Other arguments: compare-threshold, compare-list.
+
+            let stdout_isatty = atty::is(atty::Stream::Stdout);
+            let enable_text_coloring = match matches.value_of("color") {
+                Some("always") => true,
+                Some("never") => false,
+                _ => stdout_isatty,
+            };
+
+            let args = critcmp::app::Args {
+                baselines: matches.values_of_lossy("baselines").unwrap_or_default(),
```

Comment:
Parsing this causes a panic at runtime

```
thread 'main' panicked at 'Must use `Arg::allow_invalid_utf8` with `_os` lookups at `[hash: 570E2D9D9139F734]`', /.../cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.12/src/parser/matches/arg_matches.rs:1732:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This may be one of the errors that gets caught by [`debug_assert()`](https://docs.rs/clap/latest/clap/builder/struct.App.html#method.debug_assert)